### PR TITLE
[`Docs`] Add unsloth optimizations in TRL's documentation

### PR DESCRIPTION
**Pull request by younesbelkada**
# What does this PR do?

This PR adds details about unsloth library: https://github.com/unslothai/unsloth that can bring very interesting out of the box speedups to users. In the future, I'll make sure users that use unsloth + SFTTrainer have the tag `unsloth` when they push models on the hub with unsloth

cc @danielhanchen @lvwerra 

@danielhanchen let me know if the content does look to you! Feel free to add any suggestion directly in the PR

Related: https://github.com/unslothai/unsloth/issues/34

### CODE REVIEW

**Comment by danielhanchen** [comment]
Cool work @younesbelkada and thanks! Looks good - I'll add a few changes in a bit!

File: docs/source/sft_trainer.mdx [hunk 1]
```diff
@@ -410,6 +410,61 @@ We have tested NEFTune by training `mistralai/Mistral-7B-v0.1` on the [OpenAssis
 </div>
 
 Note however, that the amount of performance gain is _dataset dependent_ and in particular, applying NEFTune on synthetic datasets like [UltraChat](https://huggingface.co/datasets/stingning/ultrachat) typically produces smaller gains.
+
+### Accelerate fine-tuning using `unsloth` library
+
+You can further accelerate QLoRA / LoRA and even full-finetuning using [`unsloth`](https://github.com/unslothai/unsloth) library that is compatible with `SFTTrainer`. Currently `unsloth` supports only Llama and Mistral architectures.
```
Comment (danielhanchen) [inline docs/source/sft_trainer.mdx:416]: "You can further accelerate QLoRA / LoRA and even full-finetuning using" maybe to

You can further accelerate QLoRA / LoRA 2.2x faster and use 60% less memroy and even full-finetuning (1.1x speed boost) using [`unsloth`](https://github.com/unslothai/unsloth) library that is compatible with `SFTTrainer`. Currently `unsloth` supports only Llama (Yi, TinyLlama etc)  and Mistral architectures.

File: docs/source/sft_trainer.mdx [hunk 2]
```diff
@@ -410,6 +410,61 @@ We have tested NEFTune by training `mistralai/Mistral-7B-v0.1` on the [OpenAssis
 </div>
 
 Note however, that the amount of performance gain is _dataset dependent_ and in particular, applying NEFTune on synthetic datasets like [UltraChat](https://huggingface.co/datasets/stingning/ultrachat) typically produces smaller gains.
+
+### Accelerate fine-tuning using `unsloth` library
```
Comment (danielhanchen) [inline docs/source/sft_trainer.mdx:414]: ### Accelerate fine-tuning using `unsloth` library
maybe to

### Accelerate fine-tuning using `unsloth`

**Comment by danielhanchen** [comment]
Actually instead of comments, I just editted the docs in place here: https://github.com/huggingface/trl/compare/add-unsloth-docs...danielhanchen:trl:patch-1

**Comment by younesbelkada** [comment]
Thanks a lot @danielhanchen !  I just merged your PR with your suggestions

**Review by danielhanchen — APPROVED** [review]

**Review by danielhanchen — APPROVED** [review]

**Comment by danielhanchen** [comment]
:)

**Pull request merged**